plugins/modules/inventory_intent.py

```diff
@@ -594,11 +579,46 @@ def get_device_params(self, params):
             "serial_number": params.get("serialNumber"),
             "snmp_version": params.get("snmpVersion"),
             "type": params.get("type"),
-            "update_management_ip_list": params.get("updateMgmtIPaddressList")
+            "update_management_ip_list": params.get("updateMgmtIPaddressList"),
+            "force_sync": params.get("force_sync")
         }
 
         return device_param
 
+    def get_device_ids(self, device_ips):
+        """
+        Get the list of unique device ID's for list of specified management IP addresses of devices in Cisco DNA Center.
+        Args:
+            self (object): An instance of a class used for interacting with Cisco DNA Center.
+            device_ips (list): The management IP addresses of devices for which you want to retrieve the device ID's.
+        Returns:
+            list: The list of unique device ID's for the specified devices.
+        Description:
+            Queries Cisco DNA Center to retrieve the unique device ID associated with a device having the specified
+            IP address. If the device is not found in Cisco DNA Center, it raises an exception.
+        """
+
+        device_ids = []
+        for device_ip in device_ips:
+            try:
+                response = self.dnac._exec(
+                    family="devices",
+                    function='get_device_list',
+                    params={"managementIpAddress": device_ip}
+                )
+
+                if response:
+                    log(str(response))
+                    response = response.get("response")[0]
+                    device_id = response["id"]
+                    device_ids.append(device_id)
+
+            except Exception as e:
+                log("An error occurred while fetching the device from Cisco DNA Center")
+                raise Exception("Error while fetching device from Cisco DNA Center")
```

Comment:
can we print e as well? something like:
raise Exception("Error while fetching device from Cisco DNA Center") from e
(or)
raise Exception(f"'Error while fetching device from Cisco DNA Center' - {str(e)}")